### ISSUE TITLE
hm-rpc not sending auth header, getting "Unknown XML-RPC tag TITLE"

### ISSUE THREAD

**Issue by ctheune** [ioBroker/ioBroker.hm-rpc#1228]
**Describe the bug**

I've set up hm-rpc in docker, added the callback address, configured the port forwarding and login data. However, I'm getting the "Unkonwn XML-RPC tag TITLE" error, which afaict is a generic error if the remote side doesn't happen to send XML-RPC. 

**To Reproduce**

Not sure how to reproduce.

**Expected behavior**

The adapter instance should be starting up, connecting to my CCU.

**Screenshots & Logfiles**

Log:

<img width="1205" alt="Screenshot 2024-05-20 at 08 41 19" src="https://github.com/ioBroker/ioBroker.hm-rpc/assets/1220572/9f48ce2b-3b87-4f42-bd12-e2fad7dbe9bb">

Settings:

<img width="652" alt="Screenshot 2024-05-20 at 08 46 10" src="https://github.com/ioBroker/ioBroker.hm-rpc/assets/1220572/d79288ae-26f5-437c-8b09-0bb2f81a4a3f">

<img width="822" alt="Screenshot 2024-05-20 at 08 46 15" src="https://github.com/ioBroker/ioBroker.hm-rpc/assets/1220572/39d298b9-d521-4cff-8e36-577d7f91ddec">

tcpdump showing the missing authorization header:

<img width="1427" alt="Screenshot 2024-05-20 at 08 46 49" src="https://github.com/ioBroker/ioBroker.hm-rpc/assets/1220572/a35e645d-20aa-4834-9ffd-3009564f51df">

**Versions:**
 - hm-rpc version: 1.17.0
 - hm-rega version: 4.0
 - JS-Controller version: 5.0.19
 - Node version: v18.20.2
 - Operating system: docker on nixos
 - CCU model: CCU3
 - CCU firmware: 3.75.7

**Comment by Apollon77** [comment]
Yes, credentials are omnly used when you alsoo use HTTPS when I read the code correctly

**Comment by GermanBluefox** [comment]
Fixed in 4.1.1: username and password are also sent with XML-RPC without HTTPS. If the CCU answers with an error page instead of XML-RPC, the log now says which settings to check.
